core-bundle/contao/drivers/DC_File.php

```diff
@@ -166,19 +166,19 @@ public function edit()
 				{
 					list($key, $cls) = explode(':', $legends[$k]) + array(null, null);
 
-					$legend = "\n" . '<legend data-toggle-fieldset="' . StringUtil::specialcharsAttribute(json_encode(array('id' => $key, 'table' => $this->strTable))) . '">' . ($GLOBALS['TL_LANG'][$this->strTable][$key] ?? $key) . '</legend>';
+					$legend = "\n" . '<legend data-action="click->contao--toggle-fieldset#toggle">' . ($GLOBALS['TL_LANG'][$this->strTable][$key] ?? $key) . '</legend>';
 				}
 
-				if (isset($fs[$this->strTable][$key]))
+				if ($legend && (($fs[$this->strTable][$key] ?? null) || $cls == 'hide'))
```

Comment:
This cannot be right. Previously, the class `collapsed` was added if `$fs[$this->strTable][$key]` was set and **falsy**. Now it is added only if `$legend` is not empty and if `$fs[$this->strTable][$key]` is **not falsy**? There are two behavior changes IMHO.

Reply:
That is correct, I applied a fix here. If there is no legend, a fieldset cannot collapse, because you cannot expand it anymore. That was previously only checked in the next `elseif` adding the `hide` class.

Reply:
But why not re-use the `hide` class regardless?